conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ParamMarker.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import io.undertow.server.HttpServerExchange;
+
+/**
+ * execute arbitrary actions given a marker on a param.
+ */
+public interface ParamMarker {
+
+    void mark(Class<?> markerClass, String parameterName, Object parameterValue, HttpServerExchange exchange);
```

Comment:
this conflicts with the `[RFC] Markers and “safe” parameters in Conjure` (internal). I don't think we should cement the idea of markers being classes in this PR.

Reply:
this API opens the door for pretty wild hacks, e.g., register some marker and then mutate the exchange. is that desirable?

Reply:
1) I agree. What about providing the package and class name as strings?
2) Mutating the exchange is what we need to do for safe-logging to stash the args in an attachment. We could use a layer of abstraction to reduce the API surface but i'm not sure what it really buys us. In the case of WC, consumers will not be able to override the runtime.

Reply:
2. ok. please add plenty of code docs when implementing the stashing and the collecting, because it's going to be pretty hard to follow how things fit together.

Reply:
1: maybe just the full/canonical class name?

Reply:
1. done
2. will do when we implement it in WC